CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/panes/HeadingPaneNGTest.java

```diff
@@ -0,0 +1,146 @@
+
+    public HeadingPaneNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        // TODO Find a better solution for this. Because of this limitation these tests
+        //      will not be run on the CI server.
+
+        if (!GraphicsEnvironment.isHeadless()) {
+            // Interestingly once you throw the skip exception it doesn't call the tear down class
+            // so we need to instantiate the static mocks only once we know we will be running the
+            // tests.
+            dataAccessPreferencesMockedStatic = Mockito.mockStatic(DataAccessPreferences.class);
+
+            new JFXPanel();
+        } else {
+            throw new SkipException("This class requires the build to have a display present.");
+        }
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        dataAccessPreferencesMockedStatic.close();
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        pluginsList = Arrays.asList(dataAccessPlugin, dataAccessPlugin, dataAccessPlugin);
+
+        dataAccessPreferencesMockedStatic.reset();
+
+        headingPane = new HeadingPane(headingText, pluginsList, top, globalParamLabels);
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of getDataSources method, of class HeadingPane.
+     */
+    @Test
+    public void testGetDataSources() {
+        System.out.println("getDataSources");
+        List<DataSourceTitledPane> dataSources = headingPane.getDataSources();
+        assertEquals(dataSources.size(), 3);
+    }
+
+    /**
+     * Test of validityChanged method, of class HeadingPane, when enabled.
+     */
+    @Test
+    public void testValidityChanged_enabled() {
+        System.out.println("testValidityChanged_enabled");
+        headingPane.validityChanged(true);
+        assertTrue(headingPane.isExpanded());
+    }
+
+    /**
+     * Test of validityChanged method, of class HeadingPane, when disabled.
+     */
+    @Test
+    public void testValidityChanged_disabled() {
+        System.out.println("testValidityChanged_disabled");
+
+        String headingText = "Heading Text 2";
+        dataAccessPreferencesMockedStatic.when(() -> DataAccessPreferences.isExpanded(headingText, true)).thenReturn(false);
+
+        final HeadingPane instance = new HeadingPane(headingText, pluginsList, top, globalParamLabels);
+        instance.validityChanged(false);
+        assertFalse(instance.isExpanded());
+    }
+
+    /**
+     * Test of hierarchicalUpdate method, of class HeadingPane, when pane is not
+     * QueryEnabled.
+     */
+    @Test
+    public void testHierarchicalUpdate_paneIsNotQueryEnabled() {
+        System.out.println("testHierarchicalUpdate_paneIsNotQueryEnabled");
+
+        headingPane.hierarchicalUpdate();
+
+        verify(top, times(1)).hierarchicalUpdate();
+    }
```

Comment:
I think it would be good to add some asserts here. Currently this test only seems to check whether it got to the end of the function rather than checking anything that actually happened in it

Reply:
Not sure how to assert anything other than verify, when it's a void function, with no parameters.

Reply:
@Auriga2 Looking at the function, it appears to be making changes to fields in the class that you could probably access if you add a (protected) getter. You could then assert stuff happening to those fields.

Reply:
Yeah I can do that. I was reviewing aldebaran's changes. It looks like a different approach. Wondering what's best, or should we keep both?

Reply:
@Auriga2 the answer is probably both. It would be good to have some asserts that specific to the actual behaviour in this instance